mpitests/test_apply.py

```diff
@@ -219,6 +219,86 @@ def circuit():
     assert np.allclose(local_state_vector, local_expected_output_cpu, atol=tol, rtol=0)
 
 
+def expval_single_wire_no_param(tol, obs):
+    num_wires = numQubits
+    comm = MPI.COMM_WORLD
+    commSize = comm.Get_size()
+    num_global_wires = commSize.bit_length() - 1
+    num_local_wires = num_wires - num_global_wires
+
+    state_vector = np.zeros(1 << num_wires).astype(np.complex128)
+    local_state_vector = np.zeros(1 << num_local_wires).astype(np.complex128)
+
+    state_vector = create_random_init_state(num_wires)
+    comm.Bcast(state_vector, root=0)
+
+    comm.Scatter(state_vector, local_state_vector, root=0)
+    dev_cpu = qml.device("default.qubit", wires=num_wires, c_dtype=np.complex128)
+
+    @qml.qnode(dev_cpu)
+    def circuit():
+        qml.QubitStateVector(state_vector, wires=range(num_wires))
+        return qml.expval(obs)
+
+    expected_output_cpu = circuit()
+    comm.Bcast(expected_output_cpu, root=0)
+
+    dev_gpumpi = qml.device("lightning.gpu", wires=num_wires, mpi=True, c_dtype=np.complex128)
+
+    @qml.qnode(dev_gpumpi)
+    def circuit_mpi():
+        qml.QubitStateVector(state_vector, wires=range(num_wires))
+        return qml.expval(obs)
+
+    expected_output_gpu = circuit_mpi()
+
+    assert np.allclose(expected_output_gpu, expected_output_cpu, atol=tol, rtol=0)
+
+
+def apply_probs(tol, Wires):
+    num_wires = numQubits
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+    commSize = comm.Get_size()
+
+    state_vector = np.zeros(1 << num_wires).astype(np.complex128)
+    state_vector = create_random_init_state(num_wires)
```

Comment:
```suggestion
    state_vector = create_random_init_state(num_wires)
```

Reply:
I guess `state_vector = np.zeros(1 << num_wires).astype(np.complex128)` succeeds only since the API calls to the OS do not instantiate anything unless the memory is first accessed/manipulated. Best to remove it here and elsewhere as suggested by @vincentmr